dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/patch/factories/impl/PatchOperation.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.repository.patch.factories.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.dspace.app.rest.exception.PatchBadRequestException;
+import org.dspace.app.rest.model.RestModel;
+import org.dspace.app.rest.model.patch.Operation;
+import org.springframework.data.rest.webmvc.json.patch.LateObjectEvaluator;
+/**
+ * This patch class includes abstract and implemented methods that
+ * can be used to type check objects derived from operation values.
+ *
+ * TODO: Determine whether this type checking is needed for resource patch operations.
+ *
+ * @author Michael Spalti
+ */
+public abstract class PatchOperation<R extends RestModel, T extends Object>
+        implements ResourcePatchOperation<R> {
+
+    public abstract RestModel perform(R resource, Operation operation)
+            throws PatchBadRequestException;
+
+    /**
+     * Throws PatchBadRequestException for missing operation value.
+     * @param value the value to test
+     */
+    public void checkOperationValue(T value) {
+        if (value == null) {
+            throw new PatchBadRequestException("No value provided for the operation.");
+        }
+    }
+
+    // This is duplicated code (see org.dspace.app.rest.submit.factory.impl.PatchOperation)
+    // If it stays here, it should be DRY. Current patch resource patch operations do not
+    // use these methods.
```

Comment:
Were you able to determine if this code is needed here?  It seems like if it's not used, we shouldn't include it here until it *is* used.

Reply:
I was going to ask about this one. I definitely agree with you. I'm anticipating a need.  

These methods evaluate values that are Json objects or arrays, but as I have implemented the current patch operations values are provided as Strings. I could of course force the issue (e.g. "value": { "netid": "newNetId" }) but obviously won't -- unless there's some actual reason to do this.

Still, I'm curious if there things on the current roadmap that will require the ability to type check a json object or array when patching a resource? This is on the current todo list then perhaps these should stay.

Reply:
Since the json patch converter returns JsonValueEvaluator instances that require the type of evaluation provided by these methods, I think a complete implementation of PatchOperation requires them in principle. So at the risk of violating the YAGNI principle I think they should remain.